### DISCUSSION TITLE
Is there a limit to the sketch size of the revolving command?

### DISCUSSION THREAD

**Discussion by XiongHuyu** [ansys/pyansys-geometry#1367] [Q&A] [answered]
I got an error when using the revolving command when the sketch length is over 500 or so, I can build the model normally when the sketch length is less than 500.
![error](https://github.com/user-attachments/assets/0b4ba8ec-faf1-462e-9bac-dc7b88cee15a)

**Comment by RobPasMue** [comment]
Hi @XiongHuyu! Thanks for opening this discussion. Would it be possible for you to share the script so that I can debug it on my side as well? We have seen some space limitations from time to time. Especially when the domain space is too big. But this is a limitation of the CAD products (SpaceClaim/Discovery/Geometry Service) rather than PyAnsys Geometry.

Also, can you share as well which CAD product you are using it with?

**Comment by XiongHuyu** [comment]
Hello @RobPasMue ! I am using SCDM and here is my script.
![image](https://github.com/user-attachments/assets/5e32d3f5-17ab-45d5-a892-5dc1d17a4cb0)

**Reply by XiongHuyu** [reply]
This should work properly.

**Reply by XiongHuyu** [reply]
But when I change it, I get an error.
![1724243108471](https://github.com/user-attachments/assets/c80067ec-8101-467b-9b57-dd007c30cc6d)

**Comment by RobPasMue** [comment]
Thanks for the inputs @XiongHuyu! I'll take a look at it ASAP. In the meantime, what version of SCDM are you using? 24R1, 24R2?
And also, what version of ``ansys-geometry-core`` do you have installed? You can check this by running ``pip list`` and searching for the package.

**Reply by XiongHuyu** [reply]
@RobPasMue .i am using 24R2 of SCDM,and the version of ansys-geometry-core is 0.7.0.

**Reply by RobPasMue** [reply]
Great, thanks! I'll replicate it with the same conditions. I'll let you know as soon as I have something

**Comment by RobPasMue** [comment, marked as answer]
Hi @XiongHuyu! I just verified (sorry for the delay) and it seems that the issue is related to the modeling space available in SpaceClaim. Being the default units millimeters in SCDM, you are trying to create an object that is too big in terms of size. The following script is assuming that you are changing the default units on the client side to millimeters as well:

```py
from ansys.geometry.core import launch_modeler_with_spaceclaim
from ansys.geometry.core.sketch import Sketch
from ansys.geometry.core.misc import UNITS, Angle, DEFAULT_UNITS
from ansys.geometry.core.math import Plane, Point2D, Point3D, UNITVECTOR3D_X, UNITVECTOR3D_Y

DEFAULT_UNITS.LENGTH = UNITS.mm

m = launch_modeler_with_spaceclaim()

design = m.create_design("test")

path_radius = 5
profile_radius = 2
plane_profile = Plane(
    origin=Point3D([0, 0, 0]),
    direction_x=UNITVECTOR3D_X,
    direction_y=UNITVECTOR3D_Y,
)

profile = Sketch(plane=plane_profile)
profile.box(Point2D([250, 250]), 500, 500)
profile.plot()

design.revolve_sketch(
    "test-body",
    sketch=profile,
    axis=UNITVECTOR3D_Y,
    angle=Angle(90, UNITS.degrees),
    rotation_origin=Point3D([0, 0, 0]),
)
design.plot()

m.close()
```

The critical part that changes is the following: ``DEFAULT_UNITS.LENGTH = UNITS.mm``. This tells the Python client to use millimeters by default.

Another option is that you change the default units used in SCDM to meters instead of millimeters. Once you change that in the settings, you can also do the following so that the changes are taken into account on the client side:

```py
from ansys.geometry.core.misc import DEFAULT_UNITS

DEFAULT_UNITS.SERVER_LENGTH = UNITS.m
```

**Reply by XiongHuyu** [reply]
@RobPasMue That's great! I just tried it out and it works, thank you very much.

**Reply by RobPasMue** [reply]
Awesome! If my response served, could you mark it as an answer? This will help other users benefit from it! GitHub provides you a button right at the end of my comment to do so =)

![image](https://github.com/user-attachments/assets/4e3e13eb-0308-4ab6-a594-6853d9fcbc9d)
